src/sentry/options/manager.py

```diff
@@ -144,9 +144,11 @@ def get(self, key, silent=False):
             if result is not None:
                 # HACK(mattrobenolt): SENTRY_URL_PREFIX must be kept in sync
                 # when reading values from the database. This should
-                # be replaced by a signal.
+                # be replaced by a signal. We also want to keep the SENTRY_OPTIONS["system.url-prefix"]
+                # in sync with SENTRY_URL_PREFIX to avoid warnings
                 if key == "system.url-prefix":
                     settings.SENTRY_URL_PREFIX = result
+                    settings.SENTRY_OPTIONS["system.url-prefix"] = result
```

Comment:
Yes, if it is set before reaching here, we directly read that instead of going into the db. The default behavior is using the user configs instead of reading from the db

Reply:
Can add a test case for this! Behavior is as expected for that. The only way `settings.SENTRY_OPTIONS['system.url-prefix` is set previously is if it is in the user config files